### ISSUE TITLE
Fixed modal doesn't work with long content

### ISSUE THREAD

**Issue by egze** [fluxonui/fluxon#4]
### Fluxon UI Version

1.0.8

### Phoenix LiveView Version

1.0.0

### Current Behavior

I have a form in the modal, where I can click a link to add more fields dynamically. After a certain height - I can't scroll he modal anymore.

Should there be a way to render it not fixed?

<img width="530" alt="image" src="https://github.com/user-attachments/assets/6001392a-6a39-4fac-8e2c-b196673fd540" />

Thanks

### Expected Behavior

The modal should grow and also the whole page should be scrollable.

### Browser

Safari

### Verification

- [X] I have searched for existing issues and confirmed this is not a duplicate
- [X] I have tested this issue with the latest version of Fluxon UI
- [X] I have included all the information needed to reproduce this issue

**Comment by andrielfn** [comment]
Hey @egze, thanks for the report.

There are two main reasons/issues for that to happen: 1) by default, modals do not set a max-height, which means they will eventually grow beyond the screen limits, and 2) the modal wrapper sets an `overflow-hidden`, which means if the content is bigger than the modal limits, it will be hidden.

There are a few options to solve this, but what I would recommend is to set a `max-height` and `overflow-y-scroll` to a "content wrapper" as follows:

```
<.modal id="modal-1" class="max-w-2xl w-full p-0">
  <div class="grid grid-cols-2 gap-4 p-6 overflow-y-scroll max-h-[500px]">
    <.input name="value_1" label="Value 1" />
    <.input name="value_2" label="Value 2" />
    <.input name="value_3" label="Value 3" />
    <.input name="value_4" label="Value 4" />
    <.input name="value_5" label="Value 5" />
    <.input name="value_6" label="Value 6" />
    <.input name="value_7" label="Value 7" />
    <.input name="value_8" label="Value 8" />
    <.input name="value_9" label="Value 9" />
    <.input name="value_10" label="Value 10" />
    <.input name="value_11" label="Value 11" />
    <.input name="value_12" label="Value 12" />
    <.input name="value_13" label="Value 13" />
    <.input name="value_14" label="Value 14" />
    <.input name="value_15" label="Value 15" />
    <.input name="value_16" label="Value 16" />
    <.input name="value_17" label="Value 17" />
    <.input name="value_18" label="Value 18" />
    <.input name="value_19" label="Value 19" />
    <.input name="value_20" label="Value 20" />
  </div>
</.modal>
```

Note that I set the `overflow-y-scroll max-h-[500px]` to the div that wraps the content. The result should be something like this:

https://github.com/user-attachments/assets/6c6f3ec7-0327-40b6-96cd-cdca08ef466e

You could even play with something like `max-h-[calc(100vh-8rem)]` to make it more flexible instead of a fixed height. 

Optionally, what you could also do is set `max-h-[500px] overflow-y-scroll` directly to the modal like `<.modal class="max-h-[500px] overflow-y-scroll">`, but this would make it a bit uglier as we lose the rounded border in in the scroll:

<img width="723" alt="image" src="https://github.com/user-attachments/assets/dd38ab9f-3815-453d-9fcf-7640d040d179" />


I believe there is room to improve the default classes to handle this better. My original idea was to make it more flexible, but in the end I think it became more confusing 😅 — in any event, it's already on my radar to improve this.

Let me know if that works!

**Comment by egze** [comment]
That makes sense to make the inner content scrollable. It does have some small disadvantages, for example - my scrollbars are hidden and only appear when I scroll. Then it hurts discoverability that there's even something to scroll inside the modal.

I was more thinking about a modal that pops up in the center in the current viewport, but it doesn't do anything tricky to the page. The page remains scrollable and the modal is scrolled with it. Then you don't need to worry about the screen sizes or content height - it will always just occupy needed space. I hope I explained it, let me know if you need a small prototype.

Found a demo what I'm looking for (I think it's actually the default core_components modal 😄)

https://github.com/user-attachments/assets/98ac183b-f207-4509-a62c-10b9c372620d

**Comment by andrielfn** [comment]
That's indeed an interesting alternative! I honestly haven't considered before. I'll keep this issue open while I test it more and see if I can make this the default behavior.

Thanks for the feedback! ✨

**Comment by andrielfn** [comment]
Alright, I have something:

https://github.com/user-attachments/assets/cc20bb95-7a95-4c70-86a9-8a41a5980b42

This is indeed much better now. Again, thanks for the feedback. Available in the **v1.0.11.**
